plugins/logging/pkg/gateway/system.go

```diff
@@ -72,3 +78,93 @@ func (p *Plugin) UseKeyValueStore(client system.KeyValueStoreClient) {
 	p.uninstallController.Set(ctrl)
 	<-p.ctx.Done()
 }
+
+func (p *Plugin) watchGlobalCluster(client managementv1.ManagementClient) {
+	clusterClient, err := client.WatchClusters(p.ctx, &managementv1.WatchClustersRequest{})
```

Comment:
consider listing clusters before the for loop, to check they match the expected state on startup